tests/unit/Collectors/DBQueryCollectorTest.php

```diff
@@ -0,0 +1,143 @@
+    private const PROPRIETARY_SQL = 'NON_STANDARD * FROM user WHERE id = ?';
+
+    /** @var Application */
+    private $app;
+
+    /** @var Dispatcher */
+    private $dispatcher;
+
+    /** @var DBQueryCollector */
+    private $collector;
+
+    /** @var Agent|LegacyMockInterface|MockInterface */
+    private $agentMock;
+
+    /** @var Config|LegacyMockInterface|MockInterface */
+    private $configMock;
+
+    /** @var Connection|LegacyMockInterface|MockInterface */
+    private $connectionMock;
+
+    protected function _before(): void
+    {
+        $this->app = app(Application::class);
+        $this->dispatcher = app(Dispatcher::class);
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->agentMock = Mockery::mock(Agent::class);
+        $this->configMock = Mockery::mock(Config::class);
+        $this->connectionMock = Mockery::mock(Connection::class);
+
+        $this->connectionMock
+            ->shouldReceive('getName');
+
+        $this->collector = new DBQueryCollector(
+            $this->app,
+            $this->configMock,
+            new RequestStartTime(0.0)
+        );
+        $this->collector->useAgent($this->agentMock);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->dispatcher->forget(QueryExecuted::class);
+    }
+
+    public function testCollectorName(): void
+    {
+        self::assertEquals('query-collector', $this->collector->getName());
+    }
+
+    public function testQueryExecutedListener(): void
+    {
+        $this->configMock
+            ->shouldReceive('get')
+            ->with('elastic-apm-laravel.spans.querylog.enabled')
+            ->andReturn(true);
+
+        $this->dispatcher->dispatch(
+            new QueryExecuted(
+                self::SQL,
+                [1],
+                500,
+                $this->connectionMock
+            )
+        );
+
+        $measure = $this->collector->collect()->first();
+        self::assertEquals('SELECT user', $measure['label']);
+        self::assertGreaterThan(0.0, $measure['start']);
+        self::assertEquals(500.0, $measure['duration']);
+        self::assertEquals('db.mysql.query', $measure['type']);
+        self::assertEquals('query', $measure['action']);
+    }
+
+    public function testDisabledFastQueryExecutedListener(): void
+    {
+        $this->configMock
+            ->shouldReceive('get')
+            ->with('elastic-apm-laravel.spans.querylog.enabled')
+            ->andReturn('auto');
+
+        $this->configMock
+            ->shouldReceive('get')
+            ->with('elastic-apm-laravel.spans.querylog.threshold')
+            ->andReturn(1000.0);
+
+        $this->dispatcher->dispatch(
+            new QueryExecuted(
+                self::SQL,
+                [1],
+                500,
+                $this->connectionMock
+            )
+        );
+
+        $measure = $this->collector->collect()->first();
+        self::assertNull($measure);
```

Comment:
would that also mean that `self::assertTrue($this->collector->collect()->isEmpty();`?

Reply:
This is a cleaner assert, I will change it.